language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FieldAccessContext.java

```diff
@@ -50,10 +52,14 @@ public FieldAccessContext(Class<T> attachmentPoint) {
     protected List<LSCompletionItem> getEntries(BallerinaCompletionContext ctx,
                                                 ExpressionNode expr,
                                                 boolean optionalFieldAccess) {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
         FieldAccessCompletionResolver resolver = new FieldAccessCompletionResolver(ctx, optionalFieldAccess);
         List<Symbol> symbolList = resolver.getVisibleEntries(expr);
 
-        return this.getCompletionItemList(symbolList, ctx);
+        //Add typegurad snippet for each union type variable symbol.
+        completionItems.addAll(TypeGuardUtil.getTypeGuardDestructedItems(ctx, expr));
```

Comment:
It seems we use the `ContextTypeResolver` to capture the type of the expression. When we invoke the `FieldAccessCompletionResolver`, we already capture the required type-symbol information, but not saved it. Shall we modify `FieldAccessCompletionResolver` to capture the resolved type information? Then we can pass the FieldAccess expression (parent of expr here) and the type down the call hierarchy